src/sandstorm/supervisor.c++

```diff
@@ -905,9 +905,22 @@ void SupervisorMain::unshareOuter() {
     KJ_SYSCALL(unshare(CLONE_NEWUSER | CLONE_NEWNS | CLONE_NEWIPC | CLONE_NEWUTS | CLONE_NEWPID));
 
     // Map ourselves as 1000:1000, since it costs nothing to mask the uid and gid.
+    uid_t fakeUid = 1000;
+    gid_t fakeGid = 1000;
+
+    if (devmode) {
+      // "Randomize" the UID and GID in dev mode. This catches app bugs where the app expects the
+      // UID or GID to be always 1000, which is not true of servers that use the privileged sandbox
+      // rather than the userns sandbox. (The "randomization" algorithm here is only meant to
+      // appear random to a human.)
+      time_t now = time(nullptr);
+      fakeUid = now * 4721 % 2000 + 1;
```

Comment:
`<drew-having-fun>`

These constants can be reduced by the following equivalences:

```
now * 4721 % 2000 + 1 =
(now * 4000 + now * 721) % 2000 + 1 =
((now * 4000 % 2000) + (now * 721 % 2000)) % 2000 + 1 =
now * 721 % 2000 + 1
```

A similar proof applies to the `fakeGid` case.

```
fakeUid = now * 721 % 2000 + 1;
fakeGid = now * 791 % 2000 + 1;
```

Both 791 and 721 are coprime with 2000, so each is a generator over the group of integers between 0 and 2000.  This means they'll cover each possible UID over the course of 2000 seconds.

791 and 721 share a factor of 7, so you will see repeated (uid:gid) pairs every (7 * 113 * 103) * 2000 seconds, which could theoretically be improved if they didn't share a factor to simply their product.

`</drew-having-fun>`

No action required IMO. :)